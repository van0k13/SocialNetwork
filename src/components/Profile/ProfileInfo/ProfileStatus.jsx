import React from 'react';
import styles from './ProfileInfo.module.css';
import Preloader from '../../common/preloader/preloader';

class ProfileStatus extends React.Component {
    // if (!props.profile) {
    //     return <Preloader />
    // }
    state = {
        statusEditMode: false
    }
    activateEditMode = () => {
        this.setState({
            statusEditMode:  true
        })
    }
    deActivateEditMode = () => {
        this.setState({
            statusEditMode:  false
        })
    }
    render() {
        return (
            <>
                {!this.state.statusEditMode &&
                    <div>
                        <span onDoubleClick={this.activateEditMode}>ffd</span>
                    </div>
                }
                {this.state.statusEditMode &&
                    <div>
                        <input autoFocus={true} onBlur={this.deActivateEditMode} value={this.props.status} />
                    </div>
                }
            </>
        )
    }
}

export default ProfileStatus;