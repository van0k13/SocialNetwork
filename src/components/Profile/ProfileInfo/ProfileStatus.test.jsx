import React from 'react';
import {create} from 'react-test-renderer';
import ProfileStatus from './ProfileStatus';


describe('Profile Status Component', ()=>{
    test('status from props should be in state', ()=>{
        const component = create(<ProfileStatus status="der hund i kisulik"/>);
        const instance = component.getInstance();
        expect(instance.state.status).toBe("der hund i kisulik");
    });

    test('status creation <input> should be displayed', ()=>{
        const component = create(<ProfileStatus status="der hund i kisulik"/>);
        const root = component.root;
        expect(()=>{
            let input = root.findByType('input')
        }).toThrow();
    });

    test('status creation <span> should be displayed', ()=>{
        const component = create(<ProfileStatus status="der hund i kisulik"/>);
        const root = component.root;
        let span = root.findByType('span')
        expect(span).not.toBeNull();
    });

    test('status creation <span> should contain correct status', ()=>{
        const component = create(<ProfileStatus status="der hund i kisulik"/>);
        const root = component.root;
        let span = root.findByType('span')
        expect(span.children[0]).toBe("der hund i kisulik");
    });

    test('<input> should be displayed in editMode instead of <span>', ()=>{
        const component = create(<ProfileStatus status="der hund i kisulik"/>);
        const root = component.root;
        let span = root.findByType('span');
        span.props.onDoubleClick();
        let input = root.findByType('input');
        expect(input.props.value).toBe("der hund i kisulik");
    });

    test('callback should be called', ()=>{
        const mockCallBack = jest.fn();
        const component = create(<ProfileStatus status="der hund i kisulik" updateUserStatus={mockCallBack}/>);
        const instance = component.getInstance();
        instance.deActivateEditMode();
        expect(mockCallBack.mock.calls.length).toBe(1);
    });
});