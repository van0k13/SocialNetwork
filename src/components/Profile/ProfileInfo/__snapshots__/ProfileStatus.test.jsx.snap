// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Status Component Match 1`] = `undefined`;

exports[`Profile Status Component stat 1`] = `undefined`;

exports[`Profile Status Component status from props should be in state 1`] = `undefined`;
